test/integration/foreman_puppet/host_js_test.rb

```diff
@@ -101,6 +101,15 @@ class HostJSTest < IntegrationTestWithJavascript
         host = Host::Managed.search_for('name ~ "myhost1"').first
         assert_equal env.name, host.puppet.environment.name
       end
+
+      test 'puppet tab' do
+        host = FactoryBot.create(:host)
+
+        visit host_details_page_path(host)
+
+        find('button', text: 'Puppet').click
+        assert find('section > div.report-tab > div.pf-c-card__title').has_text? 'Puppet details'
```

Comment:
meanwhile, I see that the there's an issue with how the test environment loads the plugin's bundled JS,
I see that the plugin JS file is loaded in the new host page `head`, however the slot and fill doesn't seem to work during the test and I see the browser's console error: with the error: 
`Uncaught TypeError: Cannot read properties of undefined (reading 'call')` somewhere from webpack. 
I didn't see any other integration tests in other plugins, though we should fix it, probably not in this PR,
I would suggest to make it in a follow up

Reply:
cc @nadjaheitmann @MariaAga @amirfefer

Reply:
Sounds good.